app/featuresmodel.cpp

```diff
@@ -0,0 +1,154 @@
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "featuresmodel.h"
+
+FeaturesModel::FeaturesModel( LayersModel &lm, Loader &loader, QObject *parent )
+  : QAbstractListModel( parent ),
+    mLayersModel( lm ),
+    mLoader( loader )
+{
+}
+
+QgsQuickFeatureLayerPair FeaturesModel::getFLPairFromFeatureName( const QString &featureName )
+{
+  for ( QgsQuickFeatureLayerPair i : mFeatures )
+  {
+    if ( mLoader.featureTitle( i ) == featureName )
+      return i;
+  }
+  return QgsQuickFeatureLayerPair();
+}
+
+int FeaturesModel::rowCount( const QModelIndex &parent ) const
+{
+  // For list models only the root node (an invalid parent) should return the list's size. For all
+  // other (valid) parents, rowCount() should return 0 so that it does not become a tree model.
+  if ( parent.isValid() )
+    return 0;
+
+  return mFeatures.count();
+}
+
+QVariant FeaturesModel::data( const QModelIndex &index, int role ) const
+{
+  int row = index.row();
+  if ( row < 0 || row >= mFeatures.count() )
+    return QVariant();
+
+  if ( role < roleNames::featureTitle || role > roleNames::geometryType )
+    return QVariant();
+
+  if ( !index.isValid() )
+    return QVariant();
+
+  QgsQuickFeatureLayerPair feat = mFeatures.at( index.row() );
+
+  switch ( role )
+  {
+    case featureTitle:
+      return mLoader.featureTitle( feat );
+    case description:
+      return QVariant( QString( "description" ) );
+    case geometryType:
+      return QVariant( feat.feature().geometry().type() );
+  }
+
+  return QVariant();
+}
+
+void FeaturesModel::reloadDataFromLayerName( const QString &layerName )
+{
+  Q_UNUSED( layerName );
+
+  // We mock layerName because it is not yet implemented
+  QgsMapLayer *mockedLayer = mLayersModel.activeLayer();
+
+  if ( mockedLayer && ( mockedLayer->type() == QgsMapLayerType::VectorLayer ) )
+    this->reloadDataFromLayer( qobject_cast<QgsVectorLayer *>( mockedLayer ) );
+}
+
+void FeaturesModel::reloadDataFromLayer( QgsVectorLayer *layer )
+{
+  beginResetModel();
+  mFeatures.clear();
+
+  const int FEATURES_LIMIT = 100;
+
+  if ( layer )
+  {
+    QgsFeatureRequest req;
+
+    req.setLimit( FEATURES_LIMIT );
+
+    QgsFeatureIterator it = layer->getFeatures( req );
+    QgsFeature f;
+
+    while ( it.nextFeature( f ) )
+    {
+      mFeatures << QgsQuickFeatureLayerPair( f, layer );
+    }
+  }
+
+  endResetModel();
+}
+
+void FeaturesModel::activeProjectChanged()
+{
+  emptyData();
+}
+
+void FeaturesModel::activeMapThemeChanged( const QString &mapTheme )
+{
+  Q_UNUSED( mapTheme )
+  emptyData();
```

Comment:
True, I talked to @saberraz and he also said that we should show all `identifiable` layers no matter map theme (as it is in QGIS). I added this to next ticket #783 and handle that later. I added map theme check because without that application received segmentation fault when opening features of layers not in map theme. (again, it will be fixed in cpp models in #783 )